src/looprestoration.rs

```diff
@@ -990,383 +991,101 @@ unsafe fn sgr_mix_rust<BD: BitDepth>(
-                stride: ptrdiff_t,
-                a: *const int32_t,
-                b: *const int16_t,
-                w: libc::c_int,
-                h: libc::c_int,
-            );
-        }
-
-        dav1d_sgr_finish_filter2_16bpc_neon
-    };
-
-    const dav1d_sgr_weighted1_neon: fn_dav1d_sgr_weighted1_neon<Self> = {
-        extern "C" {
-            fn dav1d_sgr_weighted1_16bpc_neon(
-                dst: *mut <BitDepth16 as BitDepth>::Pixel,
-                dst_stride: ptrdiff_t,
-                src: *const <BitDepth16 as BitDepth>::Pixel,
-                src_stride: ptrdiff_t,
-                t1: *const int16_t,
-                w: libc::c_int,
-                h: libc::c_int,
-                wt: libc::c_int,
-                bitdepth_max: libc::c_int,
-            );
-        }
-
-        dav1d_sgr_weighted1_16bpc_neon
-    };
-
-    const dav1d_sgr_weighted2_neon: fn_dav1d_sgr_weighted2_neon<Self> = {
-        extern "C" {
-            fn dav1d_sgr_weighted2_16bpc_neon(
-                dst: *mut <BitDepth16 as BitDepth>::Pixel,
-                dst_stride: ptrdiff_t,
-                src: *const <BitDepth16 as BitDepth>::Pixel,
-                src_stride: ptrdiff_t,
-                t1: *const int16_t,
-                t2: *const int16_t,
-                w: libc::c_int,
-                h: libc::c_int,
-                wt: *const int16_t,
-                bitdepth_max: libc::c_int,
-            );
-        }
-
-        dav1d_sgr_weighted2_16bpc_neon
-    };
-
-    #[cfg(target_arch = "arm")]
-    const dav1d_wiener_filter_h_neon: fn_dav1d_wiener_filter_h_neon<Self> = {
-        extern "C" {
-            fn dav1d_wiener_filter_h_16bpc_neon(
-                dst: *mut int16_t,
-                left: *const [<BitDepth16 as BitDepth>::Pixel; 4],
-                src: *const <BitDepth16 as BitDepth>::Pixel,
-                stride: ptrdiff_t,
-                fh: *const int16_t,
-                w: intptr_t,
-                h: libc::c_int,
-                edges: LrEdgeFlags,
-                bitdepth_max: libc::c_int,
-            );
-        }
-
-        dav1d_wiener_filter_h_16bpc_neon
-    };
-
-    #[cfg(target_arch = "arm")]
-    const dav1d_wiener_filter_v_neon: fn_dav1d_wiener_filter_v_neon<Self> = {
-        extern "C" {
-            fn dav1d_wiener_filter_v_16bpc_neon(
-                dst: *mut <BitDepth16 as BitDepth>::Pixel,
-                stride: ptrdiff_t,
-                mid: *const int16_t,
-                w: libc::c_int,
-                h: libc::c_int,
-                fv: *const int16_t,
-                edges: LrEdgeFlags,
-                mid_stride: ptrdiff_t,
-                bitdepth_max: libc::c_int,
-            );
-        }
-
-        dav1d_wiener_filter_v_16bpc_neon
-    };
+    bd: BD,
+) {
+    macro_rules! asm_fn {
+        ($name:ident) => {{
+            extern "C" {
+                fn $name(
+                    dst: *mut libc::c_void,
```

Comment:
Is the `type pixel = c_void` version in scope somewhere?  The signatures of the two need to match here to make it work.

Reply:
Er, there's a typedef for it in `looprestoration.rs`, but I don't know if we have a central one. You may need to add it to `bitdepth.rs`.

Reply:
Okay.  Could we do that in a follow-up PR then? Since I think there are instances of this same thing in `looprestoration.rs`.  Also, would `pixel_erased` be a good name rather than only `pixel`?

Reply:
Yeah, followup PR is fine. And either `pixel_erased` or just `pixel` is fine.